### DISCUSSION TITLE
Filter With "remove" Option in Grouped Column Mode

### DISCUSSION THREAD

**Discussion by Menghini** [KevinVandy/material-react-table#929] [General]
I love the new "remove" feature introduced in #916, however, I can not filter anymore when the Grouped Column Mode is "remove". It would be ideal if I could still filter by those columns. Either with the global filter or the column filter.

Here is what it currently looks like with the "reorder" Grouped Column Mode
![Screenshot 2024-01-09 at 12 37 14](https://github.com/KevinVandy/material-react-table/assets/11800468/8c105118-ac15-4093-aaac-dc0c632b1c91)

And here is what it looks like with "remove" (note the column filters are gone)
![Screenshot 2024-01-09 at 12 39 14](https://github.com/KevinVandy/material-react-table/assets/11800468/bb2f4df8-6e80-43c6-b797-d8974df0c29e)

**Comment by KevinVandy** [comment]
I agree, we should still have a good default here, but how should multiple grouped columns be handled?

**Reply by Menghini** [reply]
I was thinking the same thing when I wrote this. It gets more complicated the more I think about it. There are many options:

- Obviously the global filter will work as expected, no matter the case
- In order to not make it a headache, could we set it to filter upon either the first thing that got grouped? Maybe the last thing that got grouped? Or maybe even have an option? In my specific project, I prefer to do it by the first column that was grouped. 
- Another feature loss with the "removed" is that I can no longer sort either, so this should be taken into account as well. In the example that is shared in my first post here, I have to sort by State first, then start to group it. Or maybe if it's sorted, you sort by the first state as primary and the the 2nd group as secondary, etc.

Thoughts?

**Reply by KevinVandy** [reply]
For your own implementation, you should be able to achieve what you need to my overriding the `Header` in `displayColumnDefOptions:.'mrt-row-expand'` like in this example:
https://www.material-react-table.com/docs/examples/customized-grouping

Just import MRT_TableHeadCellFilterContainer with the `header` that you want to act as the filter.
